packages/shared/src/components/profile/TopReaderWidget.tsx

```diff
@@ -0,0 +1,112 @@
+import { useQuery } from '@tanstack/react-query';
+import React, { ReactElement } from 'react';
+import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
+import { fetchTopReaders } from '../../lib/topReader';
+import { disabledRefetch } from '../../lib/func';
+import {
+  ActivityContainer,
+  ActivitySectionHeader,
+  ActivitySectionSubTitle,
+} from './ActivitySection';
+import { topReaderBadgeDocs } from '../../lib/constants';
+import { MedalBadgeIcon } from '../icons';
+import { IconSize } from '../Icon';
+import { BadgeIconGoldGradient } from '../badges/BadgeIcon';
+import {
+  Typography,
+  TypographyColor,
+  TypographyType,
+} from '../typography/Typography';
+import { formatDate, TimeFormatType } from '../../lib/dateFormat';
+import type { PublicProfile } from '../../lib/user';
+import { ClickableText } from '../buttons/ClickableText';
+
+export const TopReaderWidget = ({
+  user,
+}: {
+  user: PublicProfile;
+}): ReactElement => {
+  const { data: topReader, isLoading: isTopReaderLoading } = useQuery({
+    queryKey: generateQueryKey(RequestKey.TopReaderBadge, user, 'latest'),
+    queryFn: async () => {
+      return await fetchTopReaders(5, user.id);
+    },
+    staleTime: StaleTime.OneHour,
+    ...disabledRefetch,
+  });
+
+  if (isTopReaderLoading || topReader.length === 0) {
+    return null;
+  }
+
+  return (
+    <ActivityContainer>
+      <ActivitySectionHeader className="flex-wrap" title="Badges">
+        <ActivitySectionSubTitle className="w-full">
+          Understand more how this is computed in
+          <ClickableText
+            tag="a"
+            target="_blank"
+            className="ml-1"
+            href={topReaderBadgeDocs}
+          >
+            daily.dev docs
+          </ClickableText>
+        </ActivitySectionSubTitle>
+      </ActivitySectionHeader>
+
+      <div className="w-60 rounded-10 border border-border-subtlest-tertiary p-3">
+        <div className="flex">
+          <MedalBadgeIcon
+            secondary
+            size={IconSize.XLarge}
+            fill="url(#goldGradient)"
+          />
+          <BadgeIconGoldGradient />
+          <div className="flex flex-col">
+            <Typography
+              type={TypographyType.Body}
+              color={TypographyColor.Primary}
+              bold
+            >
+              X{topReader[0]?.total ?? 0}
+            </Typography>
+            <Typography
+              type={TypographyType.Footnote}
+              color={TypographyColor.Tertiary}
+            >
+              Top reader
+            </Typography>
+          </div>
+        </div>
+
+        {topReader.length > 0 && (
```

Comment:
Is it possible to be `null`? Like even at initial render before isFetched is true?
```suggestion
        {topReader?.length > 0 && (
```

Reply:
There's this at the top, so it shouldn't render at all if the array is empty.
I'm going to remove this check, as it's redundant.
```
if (isTopReaderLoading || topReaders?.length === 0) {
  return null;
}
```